### PR TITLE
Louise dev

### PR DESCRIPTION
**Pull request by lamr18**
**Summary:**

-  fiona_gen.py is used to generate one oxDNA configuration in the form of a .conf and a .top file. louise_gen.py expands this to generate multiple configurations from one command.

- louise_gen.py requires 3 sets of 3 parameters:
    - overall length of the polymer (-n), requires input of start length, step size and end length
    - dsDNA section length (-ds), requires input of start length, step size and end length
    - ssDNA section length: 10 (-ss), requires input of start length, step size and end length

- Code goes though all the combinations of total length, ssDNA section length and dsDNA section length and outputs a folder containing the oxdna.out.top and oxdna.out.conf files for each combination.

- The output is a  folder called "oxDNA_sims" which  contains folders "sim1", "sim2", etc. with each set of top and conf files in them.

- Command format is "python louise_gen.py -n start_length step_size end_length -ds start_length step_size end_length -ss start_length step_size end_length"
Example format is "python louise_gen.py -n 30 10 100 -ds 10 5 50 -ss 5 5 30 -f out". This command will output 432 folders, each containing a .top and a .conf file.

- The 'write_oxDNA_folder attribute was added to the class 'System'. Changes to the original system.py file were avoided using a copy of the system.py file in the block-dna directory.

### CODE REVIEW

**Comment by debeshmandal** [comment]
Hi @lamr18 - I've just had a look through your code and spotted a trick that I think you can do to avoid copy and pasting the whole system.py file to your folder. You can subclass the original `System` class and add your method to the subclass. This would look like the following:

```python
from origamiUROP.oxdna import System

class SystemWithFolder(System):
    def __init__(self, *args, **kwargs):
        super()

    def write_oxDNA_folder(self, ...):
        ...
        return
```
I recommend changing this in your protocols folder because it will mean that your new class will stay up to date with any further changes we make.

Also, please can you add `.DS_Store` to the `.gitignore` file so these files should not be present in the repository (I think, unless you think they are in which case let me know)

File: protocols/block-dna/fiona_gen.py [hunk 1]
```diff
@@ -8,7 +8,8 @@
 
 import numpy as np
 
-from origamiUROP.oxdna import System, Strand, Nucleotide
+from origamiUROP.oxdna import Strand, Nucleotide
+from system import System
```
Comment (debeshmandal) [inline protocols/block-dna/fiona_gen.py:12]: see comment about subclassing

File: protocols/block-dna/system.py [hunk 2]
```diff
@@ -0,0 +1,317 @@
+"""
```
Comment (debeshmandal) [inline protocols/block-dna/system.py:1]: see comment about subclassing

**Review by debeshmandal — CHANGES_REQUESTED** [review]
See comments about subclassing and also removing `.DS_Store` files and making sure they're present in `.gitignore`

**Comment by lamr18** [comment]
Hi @debeshmandal,
Thanks for your feedback, I've changed the system.py file in protocols so that it contains a subclass 'SystemWithFolder'. 
For your request about .DS_Store files and .gitignore , I've never dealt with those files but I found this : [https://stackoverflow.com/questions/107701/how-can-i-remove-ds-store-files-from-a-git-repository](url). Should I just follow the instructions in the top answer? 
Thanks a lot !
-Louise

**Comment by debeshmandal** [comment]
No worries, I will explain now:

The file `.gitignore` contains a list of files that git should ignore when looking through the directory structure of the repository.

We can add files and regex patterns to ensure that git does not add files that are not useful to the repository, for example files that are generated and take up large amounts of disk space.

In this case, `.DS_STORE` is one of these files so we can just append its name to the end of the `.gitignore` file

**Comment by lamr18** [comment]
Thanks for your explanation! I think I correctly removed the .DS_Store files and added it to the .gitignore :)

**Pull request merged**